include/swift/SIL/SILInstruction.h

```diff
@@ -3787,6 +3787,12 @@ class TermInst : public SILInstruction {
 
   bool isBranch() const { return !getSuccessors().empty(); }
 
+  /// Returns true if this terminator exits the function.
+  bool isFunctionExiting() const {
+    return getKind() == ValueKind::ThrowInst ||
+           getKind() == ValueKind::ReturnInst;
```

Comment:
> Can this be an exhaustive switch
Makes sense. I'll change it.

> Should Unreachable also be considered to exit the function?
Not for the purposes where this function is used.